### PR TITLE
Wip/hoop chart

### PR DESCRIPTION
**Pull request by webwarrior-ws**
Addresses #172 (Better design for cheese chart (so that text is inside it))

### CODE REVIEW

**Comment by knocte** [comment]
@webwarrior-ws good work! Can you enable GitHubActions in your fork please?

File: src/GWallet.Frontend.XF/HoopChart.fs [hunk 1]
```diff
@@ -185,40 +173,48 @@ module HoopChart =
...
 
+    let balanceLabel = 
+        Label( 
+            Text = "...", 
+            HorizontalTextAlignment = TextAlignment.Center,
+            FontSize = 25.)
+            
+    let balanceTagLabel = 
+        Label( 
+            Text = "Amount Balance",
```
Comment (knocte) [inline src/GWallet.Frontend.XF/HoopChart.fs:187]: @webwarrior-ws actually let's have the same text as the old label: `Total Assets:\n`

**Comment by webwarrior-ws** [comment]
Changed label text. CI job failure seems to be unrelated.

**Comment by knocte** [comment]
> CI job failure seems to be unrelated.

True, it was unrelated, but do you mind rebasing your PR? I fixed the CI issue in `wip/hoopChart` branch.

**Comment by webwarrior-ws** [comment]
> > CI job failure seems to be unrelated.
> 
> True, it was unrelated, but do you mind rebasing your PR? I fixed the CI issue in `wip/hoopChart` branch.

I must have done something wrong, because now I can't push:

> Pushing to https://github.com/webwarrior-ws/geewallet
> To https://github.com/webwarrior-ws/geewallet
> ! [remote rejected]   wip/hoopChart -> wip/hoopChart (refusing to allow an OAuth App to create or update workflow > `.github/workflows/CI.yml` without `workflow` scope)
> error: failed to push some refs to 'https://github.com/webwarrior-ws/geewallet'

**Comment by knocte** [comment]
If that doesn't do it; it seems it's the way you cloned your fork? I recommend you to use SSH, not weird OAuth stuff.

**Comment by webwarrior-ws** [comment]
OK, now with SSH it worked

**Comment by webwarrior-ws** [comment]
> I said rebase, not merge :(
> 
> When rebasing, there should not be any merge commits on top of your branch.

So now what? Do I checkout or reset to latest commit that is not mine?

**Comment by knocte** [comment]
IMO you should first delete your branch locally, then checkout commit 9e70c433903c475096354bf64c6c663d30407298 and create a new branch from it (same name as the one that this PR is targetting), then cherry-pick your commits from github, then push with --force.

**Comment by knocte** [comment]
Finally, I had time to test it today (sorry, yesterday I was in the eye-doctor and they gave me something that made my sight blurry).

Code-wise, the proposed PR looks fine; integration seems clean. But I have this rendering problem:

Screenshot 2022-07-07 at 9 03 13 PM

As you can see, the circle is too big, and causes the currency frames to need scrolling. Ideally the circle shouldn't be rendered as big and everything should fit. Do you know if this is easy to fix?

**Comment by webwarrior-ws** [comment]
> Finally, I had time to test it today (sorry, yesterday I was in the eye-doctor and they gave me something that made my sight blurry).
> 
> Code-wise, the proposed PR looks fine; integration seems clean. But I have this rendering problem:
> 
> As you can see, the circle is too big, and causes the currency frames to need scrolling. Ideally the circle shouldn't be rendered as big and everything should fit. Do you know if this is easy to fix?

Probably not. 
In order to do that, shapes that make up hoop chart should resize to fill the container. And honestly I don't know how shapes and paths in Xamarin.Froms work.
Also the chart can't be allowed to get too small, because label might overlap with the shapes.

**Comment by knocte** [comment]
Superseded by https://gitlab.com/nblockchain/geewallet/-/merge_requests/141

**Pull request closed without merging**